projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -4328,14 +4328,29 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     }
 
     /**
-     * Manually marks the `IgxGridComponent` for change detection.
+     * Triggers change detection for the `IgxGridComponent`.
+     * Calling markForCheck also triggers the grid pipes explicitly, resulting in all updates being processed.
+     * May degrade performance if used when not needed, or if misused:
+     * ```typescript
+     * // not needed to use, because the grid updates itself succesfully when a primitive has changed:
+     * grid.data.forEach(rec => {
+     *  rec = newValue;
+     *  grid.markForCheck();
+     * });
```

Comment:
This is misleading. The grid does not check references inside the data source array.
So you should call `markForCheck()` again and preferably outside of the `forEach` loop

Reply:
That was meant to be the "Dont this at home" example

Reply:
I will try to make it more clear which example is "DO it" and which "Dont't do it"